documentation/source/quickstart.rst

```diff
@@ -4,291 +4,188 @@
+   :class:`cocotb.clock.Clock`.
+   No need to write your own clock generator!
 
+   You would start :class:`~cocotb.clock.Clock` with
+   ``cocotb.fork(Clock(dut.clk, 1, units="ns").start())`` near the top of your test,
+   after importing it with ``from cocotb.clock import Clock``.
 
-.. _quickstart_reading_values:
 
-Reading values from signals
----------------------------
+.. _quickstart_running_a_test:
 
-Values in the DUT can be accessed with the :attr:`~cocotb.handle.NonHierarchyObject.value`
-property of a handle object.
-A common mistake is forgetting the ``.value`` which just gives you a reference to a handle
-(useful for defining an alias name), not the value.
+Running a Test
+==============
 
-The Python type of a value depends on the handle's HDL type:
+In order to run a test,
+you create a ``Makefile`` that contains information about your project
+(i.e. the specific DUT and test),
+and at the end include a cocotb-provided makefile that handles building the project's design sources
+and running a simulation.
 
-* Arrays of ``logic`` and subtypes of that (``sfixed``, ``unsigned``, etc.) are of type :class:`~cocotb.binary.BinaryValue`.
-* Integer nets and constants (``integer``, ``natural``, etc.) return :class:`int`.
-* Floating point nets and constants (``real``) return :class:`float`.
-* Boolean nets and constants (``boolean``) return :class:`bool`.
-* String nets and constants (``string``) return :class:`bytes`.
+In the ``Makefile`` shown below we specify:
 
-For a :class:`~cocotb.binary.BinaryValue` object, any unresolved bits are preserved and
-can be accessed using the :attr:`~cocotb.binary.BinaryValue.binstr` attribute,
-or a resolved integer value can be accessed using the :attr:`~cocotb.binary.BinaryValue.integer` attribute.
+* the default simulator to use (:make:var:`SIM`),
+* the default language of the toplevel module or entity (:make:var:`TOPLEVEL_LANG`, ``verilog`` in our case),
+* the design source files (:make:var:`VERILOG_SOURCES` and :make:var:`VHDL_SOURCES`),
+* the toplevel module or entity to instantiate (:envvar:`TOPLEVEL`,  ``my_design`` in our case),
+* and a Python module that contains our cocotb tests (:envvar:`MODULE`, ``test_my_design`` in our case).
 
-.. code-block:: pycon
+.. code-block:: makefile
 
-    >>> # Read a value back from the DUT
-    >>> count = dut.counter.value
-    >>> print(count.binstr)
-    1X1010
-    >>> # Resolve the value to an integer (X or Z treated as 0)
-    >>> print(count.integer)
-    42
-    >>> # Show number of bits in a value
-    >>> print(count.n_bits)
-    6
+    # This is file Makefile
 
-We can also cast the signal handle directly to an integer:
+    # defaults
+    SIM ?= icarus
+    TOPLEVEL_LANG ?= verilog
 
-.. code-block:: pycon
+    # assuming there are two implementations of the same design
+    # in the directory where this Makefile is located
+    VERILOG_SOURCES += $(PWD)/my_design.sv
+    VHDL_SOURCES += $(PWD)/my_design.vhdl
```

Comment:
Either choose the verilog one, or the VHDL one. You can add a comment like "Use VHDL_SOURCES instead for VHDL designs."